lib/zk_recipes/cache.rb

```diff
@@ -182,35 +184,33 @@ def update_cache(path)
       @cache[path] = CachedPath.new(value, stat: stat, valid: valid)
 
       ActiveSupport::Notifications.instrument(AS_NOTIFICATION, instrument_params)
-      debug { "update_cache path=#{path} raw_value=#{raw_value.inspect} value=#{value.inspect} stat=#{stat.inspect}" }
+      @logger&.debug { "update_node path=#{path} raw_value=#{raw_value.inspect} value=#{value.inspect} stat=#{stat.inspect}" }
       true
     rescue ::ZK::Exceptions::ZKError => e
-      warn { "update_cache path=#{path} exception=#{e.inspect}, retrying" }
+      @logger&.warn { "path=#{path} exception=#{e.inspect}, retrying" }
       retry
     rescue ::ZK::Exceptions::KeeperException, ::Zookeeper::Exceptions::ZookeeperException => e
-      error { "update_cache path=#{path} exception=#{e.inspect}" }
+      @logger&.error { "path=#{path} exception=#{e.inspect}" }
       false
     end
 
     def process_pending_updates
-      return if @pending_updates.empty?
-      debug { "processing pending updates=#{@pending_updates.size}" }
+      if @pending_updates.empty?
+        @logger&.debug("process_pending_updates: no pending updates")
+        return
+      end
+
+      unless !@zk.connected?
```

Comment:
Odd negation, `if @zk.connected?` may be clearer

Reply:
Good catch. Fixed.